hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/PersistentIOEngine.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.io.hfile.bucket;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.hadoop.util.Shell;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A class implementing PersistentIOEngine interface supports file integrity verification
+ * for {@link BucketCache} which use persistent IOEngine
+ */
+@InterfaceAudience.Private
+public abstract class PersistentIOEngine implements IOEngine {
+  private static final Logger LOG = LoggerFactory.getLogger(PersistentIOEngine.class);
+  private static final DuFileCommand DU = new DuFileCommand(new String[] {"du", ""});
+  protected final String[] filePaths;
+
+  public PersistentIOEngine(String[] filePaths) {
```

Comment:
Why the arg not using String... ?  U can just call super(filePath); from FileMmapIOEngine ?

Reply:
Because FileIOEngine can use multiple files as cache file.

Reply:
I dont mean used String type.  Strings only but as in FileIOE..  String... filePaths
If so in FileMmapIOEngine  we can just call super(filePath)?

Reply:
OK, I see.